src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Parameters/SearchParameterValidator.cs

```diff
@@ -0,0 +1,121 @@
+            using (IScoped<IFhirOperationDataStore> fhirOperationDataStore = _fhirOperationDataStoreFactory())
+            {
+                (var activeReindexJobs, var reindexJobId) = await fhirOperationDataStore.Value.CheckActiveReindexJobsAsync(cancellationToken);
+                if (activeReindexJobs)
+                {
+                    throw new JobConflictException(string.Format(Resources.ChangesToSearchParametersNotAllowedWhileReindexing, reindexJobId));
+                }
+            }
+
+            var validationFailures = new List<ValidationFailure>();
+
+            if (string.IsNullOrEmpty(searchParam.Url))
+            {
+                validationFailures.Add(
+                    new ValidationFailure(nameof(Base.TypeName), Resources.SearchParameterDefinitionInvalidMissingUri));
+            }
+            else
+            {
+                try
+                {
+                    _searchParameterDefinitionManager.GetSearchParameter(new Uri(searchParam.Url));
+
+                    // If a post, then it is a creation of a new search parameter
+                    // only allow this if no other parameters exist with the same Uri
+                    if (method.Equals(HttpPostName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        // if no exception is thrown, then the search parameter with the same Uri was found
+                        // and this is a conflict
+                        validationFailures.Add(
+                            new ValidationFailure(
+                                nameof(searchParam.Url),
+                                string.Format(Resources.SearchParameterDefinitionDuplicatedEntry, searchParam.Url)));
+                    }
+                }
+                catch (SearchParameterNotSupportedException)
+                {
+                    // if thrown, then the search parameter is not found
+                    // if a PUT, then we should be updating an exsting paramter, but it was not found
+                    if (method.Equals(HttpPutName, StringComparison.OrdinalIgnoreCase) ||
+                        method.Equals(HttpDeleteName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        // if an exception above was thrown, then the search parameter with the same Uri was not found
+                        // and DELETE or PUT can only run on existing parameter
+                        validationFailures.Add(
+                            new ValidationFailure(
+                                nameof(searchParam.Url),
+                                string.Format(Resources.SearchParameterDefinitionNotFound, searchParam.Url)));
+                    }
+                }
+            }
+
+            // validate that the url does not correspond to a search param from the spec
+            // TODO: still need a method to determine spec defined search params
+
+            // validation of the fhir path
```

Comment:
For SQL, we'll also need additional validation for composite parameters, ensuring that it's one of the existing type combinations.

Reply:
New work item created to track this validation need (78059).